tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -100,6 +101,24 @@ static void CreateOperationResultFromOutputs(
                                    ppResult);
 }
 
+struct DefaultFPEnvScope
+{
+  // _controlfp_s is non-standard and <cfenv>.feholdexceptions doesn't work on windows...?
+#ifdef _WIN32
+  unsigned int previousValue;
+  DefaultFPEnvScope() {
+    // No exceptions, preserve denormals & round to nearest.
+    _controlfp_s(&previousValue, _MCW_EM | _DN_SAVE | _RC_NEAR, _MCW_EM | _MCW_DN | _MCW_RC);
```

Comment:
Why not pass 0 as mask value to get the complete fp control word as-is in `previousValue`?
`_controlfp_s(&previousValue, _MCW_EM | _DN_SAVE | _RC_NEAR, 0);`

Reply:
I hadn't seen that was possible. Note that with 0, the set mask is ignored, so I would need a second call to modify the state.. and presumably use UINT_MAX as the mask to restore the state? Since I'm not sure want to touch the infinity and precision control bits (x87-specific), I think my current way does what we want.

Reply:
We synced up offline, it seems ok here to only restore masks that we care about.